### ISSUE TITLE
Define a set of tokens our lexer will produce

### ISSUE THREAD

**Issue by DragonAxe** [DragonAxe/CSE423-Compiler-Collective#34]
For example, how granular do we want our tokens to be?

Example 1:
```
<INT_KEYWORD, "int">
<IDENTIFIER, "main">
<PUNCTUATOR_LEFT_BRACKET, "{">
<INTEGER_CONSTANT, "28273">
<FLOATING_CONSTANT, "2183.53">
```
or Example 2:
```
<KEYWORD, "int">
<IDENTIFIER, "main">
<PUNCTUATOR, "{">
<CONSTANT, "28273">
<CONSTANT, "2183.53">
```
or Somewhere in-between?

**Comment by Wmyers559** [comment]
My initial feeling is somewhere in between--for example, keep specifying the type of constant (like `INTEGER_CONSTANT`), but maybe leave keywords/punctuators as single groups. For punctuators/keywords, that verbose title just duplicates the information of the token found. For constants, that information (the type) is only part of the token, so it makes sense to pull that out.

**Comment by Wmyers559** [comment]
Closed due to the implementation being finalized and merged in #35
